### PR TITLE
DIT-427 Send Email on BTI Submission

### PR DESCRIPTION
**Pull request by ed0906**

### CODE REVIEW

**Comment by ed0906** [comment]
Dont Merge this yet as it depends on https://github.com/hmrc/hmrc-email-renderer/pull/310

File: conf/application.conf [hunk 1]
```diff
@@ -109,6 +109,11 @@ microservice {
         port = 9852
       }
 
+      email {
```
Comment (gokyo) [inline conf/application.conf]: we need it also in `app-config-base`

File: app/models/Email.scala [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  override val templateId: String = EmailType.APPLICATION_SUBMITTED.toString
+}
+
+object ApplicationSubmittedEmail {
+  implicit val format: OFormat[ApplicationSubmittedEmail] = Json.format[ApplicationSubmittedEmail]
+}
+
+case class ApplicationSubmittedParameters
+(
+  recipientName_line1: String, // Must match hrmc-email-renderer SalutationHelper param name The full name of the recipient
```
Comment (gokyo) [inline app/models/Email.scala]: can you make the comment more readable?

File: test/unit/connectors/EmailConnectorSpec.scala [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      stubFor(post(urlEqualTo("/hmrc/email"))
+          .withRequestBody(new EqualToJsonPattern(fromResource("advice_request_email-request.json"), true, false))
+        .willReturn(aResponse()
+          .withStatus(HttpStatus.SC_ACCEPTED))
+      )
+
+      val email = ApplicationSubmittedEmail(Seq("user@domain.com"), ApplicationSubmittedParameters("ref", "name"))
+
+      await(connector.send(email))
+    }
```
Comment (gokyo) [inline test/unit/connectors/EmailConnectorSpec.scala]: before we forget.... 
can you add this?

```
verify(
        postRequestedFor(urlEqualTo("/hmrc/email"))
          .withoutHeader("X-Api-Token")
      )
```

**Comment by gokyo** [comment]
Let's merge this PR before #83

File: test/unit/connectors/EmailConnectorSpec.scala [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import org.mockito.BDDMockito._
+import org.scalatest.mockito.MockitoSugar
+import play.api.Environment
+import play.api.libs.ws.WSClient
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.audit.DefaultAuditConnector
+import uk.gov.hmrc.play.bootstrap.http.DefaultHttpClient
+import uk.gov.hmrc.play.test.{UnitSpec, WithFakeApplication}
+
+class EmailConnectorSpec extends UnitSpec
```
Comment (gokyo) [inline test/unit/connectors/EmailConnectorSpec.scala]: This needs refactoring (see #83)

We will need this class to extend `TestConnector`

File: conf/application.conf (hunk 1 above)
Comment (ed0906) [reply conf/application.conf]: already there

File: app/models/Email.scala (hunk 2 above)
Comment (ed0906) [reply app/models/Email.scala]: Done

File: test/unit/connectors/EmailConnectorSpec.scala (hunk 4 above)
Comment (SteveSugden) [reply test/unit/connectors/EmailConnectorSpec.scala]: I'm re-factoring that.  Do you mean `ConnectorTest`?

File: test/unit/connectors/EmailConnectorSpec.scala [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  override protected def beforeEach(): Unit = {
+    super.beforeEach()
+
+    given(configuration.emailUrl).willReturn(wireMockUrl)
+  }
+
+  "Connector 'Send'" should {
+    "POST Email payload" in {
+      stubFor(post(urlEqualTo("/hmrc/email"))
+          .withRequestBody(new EqualToJsonPattern(fromResource("advice_request_email-request.json"), true, false))
```
Comment (SteveSugden) [inline test/unit/connectors/EmailConnectorSpec.scala]: Ed - did you forget to check in this resource file?

Comment (SteveSugden) [reply test/unit/connectors/EmailConnectorSpec.scala]: I pushed a small change to this branch @ed0906 - (5d7f7f9)

File: test/unit/connectors/EmailConnectorSpec.scala [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        .willReturn(aResponse()
+          .withStatus(HttpStatus.SC_ACCEPTED))
+      )
+
+      val email = ApplicationSubmittedEmail(Seq("user@domain.com"), ApplicationSubmittedParameters("ref", "name"))
+
+      await(connector.send(email))
```
Comment (gokyo) [inline test/unit/connectors/EmailConnectorSpec.scala]: `shouldBe ((): Unit)`

File: test/unit/connectors/EmailConnectorSpec.scala (hunk 5 above)
Comment (ed0906) [reply test/unit/connectors/EmailConnectorSpec.scala]: Apologies, cheers @SteveSugden !

File: test/unit/connectors/EmailConnectorSpec.scala (hunk 6 above)
Comment (ed0906) [reply test/unit/connectors/EmailConnectorSpec.scala]: Does it matter? If you return unit it usually means you dont care about the response

**Pull request merged**